actionpack/lib/action_dispatch/middleware/host_authorization.rb

```diff
@@ -17,7 +17,17 @@ module ActionDispatch
   # if +config.consider_all_requests_local+ is set to true, otherwise the body is empty.
   class HostAuthorization
     ALLOWED_HOSTS_IN_DEVELOPMENT = [".localhost", IPAddr.new("0.0.0.0/0"), IPAddr.new("::/0")]
-    PORT_REGEX = /(?::\d+)?/.freeze
+    PORT_REGEX = /(?::\d+)?/ # :nodoc:
+    IPV6_HOSTNAME = /(?<host>[a-f0-9]*:[a-f0-9.:]+)/i # :nodoc:
+    IPV4_HOSTNAME = /(?<host>\d+.\d+.\d+.\d+)/i # :nodoc:
```

Comment:
Also, these dots should be escaped:

```suggestion
    IPV4_HOSTNAME = /(?<host>\d+[.]\d+[.]\d+[.]\d+)/i # :nodoc:
```

Or:

```suggestion
    IPV4_HOSTNAME = /(?<host>\d+\.\d+\.\d+\.\d+)/i # :nodoc:
```